src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/net462/ConfigurationBinder/Bind.generated.txt

```diff
@@ -137,7 +137,7 @@ namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
 
             if (configuration["MyInt"] is string value1)
             {
-                instance.MyInt = ParseInt(value1, () => configuration.GetSection("MyInt").Path);
+                instance.MyInt = ParseInt(value1, configuration.GetSection("MyInt").Path);
```

Comment:
My understanding is that this change is trading off closure allocation for delayed execution. Not super familiar with this code, but is there any chance that the eagerly evaluated expression could throw an exception and introduce regressions?

From my perspective it might be preferable if the `ParseInt` helper accepted the `configuration` parameter and an optional suffix parameter for `"MyInt"` so that `GetSection` is only evaluated in case of exception.

cc @tarekgh

Reply:
Although this is a good idea, but I want to limit the change here as this is not scoped to `ParseInt` only but all parse methods. I prefer for now keep passing the string.